src/navigation/models.ts

```diff
@@ -0,0 +1,91 @@
+import {MouseEventHandler} from 'react';
+
+import {
+    NavigationItemData,
+    NavigationItemModel,
+    NavigationLinkItem,
+    ThemedNavigationLogoData,
+} from '../models';
+
+export interface MobileMenuButtonProps {
+    isSidebarOpened: boolean;
+    onSidebarOpenedChange: (arg: boolean) => void;
+}
+
+export enum ItemColumnName {
+    Left = 'left',
+    Right = 'right',
+    Top = 'top',
+    Bottom = 'bottom',
+}
+
+export enum NavigationLayout {
+    Desktop = 'desktop',
+    Mobile = 'mobile',
+    Dropdown = 'dropdown',
+}
+
+export interface NavigationItemProps {
+    data: NavigationItemData;
+    className?: string;
+    onClick?: MouseEventHandler;
+    isActive?: boolean;
+    isTopLevel?: boolean;
+    menuLayout?: NavigationLayout;
+    hidePopup?: () => void;
+}
+
+export interface NavigationListItemProps extends Pick<NavigationItemProps, 'menuLayout'> {
+    data: NavigationItemModel;
+    column: ItemColumnName;
+    index: number;
+    activeItemId?: string;
+    className?: string;
+    onActiveItemChange: (id?: string) => void;
+}
+
+export interface NavigationListProps
+    extends Pick<
+        NavigationListItemProps,
+        'activeItemId' | 'column' | 'className' | 'onActiveItemChange' | 'menuLayout'
```

Comment:
there are `activeItemId` and `onActiveItemChange` repeating in interface below, could we create some base interface?

Reply:
done